cpp/src/parquet/arrow/arrow_schema_test.cc

```diff
@@ -644,6 +649,78 @@ TEST_F(TestConvertParquetSchema, ParquetRepeatedNestedSchema) {
   ASSERT_NO_FATAL_FAILURE(CheckFlatSchema(arrow_schema));
 }
 
+TEST_F(TestConvertParquetSchema, ColumnSubselection) {
+  std::vector<NodePtr> parquet_fields;
+  std::vector<std::shared_ptr<Field>> arrow_fields;
+  {
+    //   optional int32 leaf1;
+    //   repeated group outerGroup {
+    //     optional int32 leaf2;
+    //     repeated group innerGroup {
+    //       optional int32 leaf3;
+    //     }
+    //   }
+    parquet_fields.push_back(
```

Comment:
are these intentionally not using logical annotations (i.e. LIST?).  Maybe ad a comment if so.

Reply:
On consideration, I think I'll scrap the new overload of `FromParquetSchema`. It's straightforward to retrieve the specific fields by index from a "whole" arrow schema and doesn't currently warrant a dedicated utility.